recipes/libdeflate/all/conanfile.py

```diff
@@ -60,51 +56,65 @@ def configure(self):
         except Exception:
             pass
 
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
     def build_requirements(self):
         if self._settings_build.os == "Windows" and not is_msvc(self):
-            if "CONAN_BASH_PATH" not in Environment().vars(self, scope="build").keys():
+            self.win_bash = True
+            if not self.conf.get("tools.microsoft.bash:path", default=False, check_type=str):
                 self.tool_requires("msys2/cci.latest")
 
     def source(self):
-        get(self, **self.conan_data["sources"][self.version], destination=self._source_subfolder, strip_root=True)
+        get(self, **self.conan_data["sources"][self.version], destination=self.source_folder, strip_root=True)
+
+    def generate(self):
+        env = VirtualBuildEnv(self)
+        env.generate()
+
+        if is_msvc(self) or self._is_clangcl:
+            vc = VCVars(self)
```

Comment:
Does `VirtualBuildEnv` automatically handle VCVars?

Reply:
I don't think so.

https://docs.conan.io/en/latest/reference/conanfile/tools/microsoft.html?highlight=vcvars#msbuild

It only handles msbuild stuff. VirtualBuildEnv is more related to environment variables.

Reply:
How does building most packages work then, when I'm not in a Visual Studio Developer Command Prompt? I figured they would fail.

Reply:
The msbuild helper does that step 😉

Reply:
@jwillikers one of the reasons I want to upgrade these recipes is so I can build everything in my git-bash-SDK-msys terminal on windows.  And so far, the new recipes seem to be doing it well.